frontend/src/components/Icon.tsx

```diff
@@ -1,36 +1,75 @@
 import * as React from 'react';
-import ReactSVG from 'react-svg';
 import './Icon.css';
+import { getSVGShadowRoot, W3SVG } from '../utils';
 
-export interface Props {
-  src: string;
-  alt?: string;
-  className?: string;
-  onClick?: () => void;
+export namespace Icon {
+  export interface Props {
+    src: string;
+    className?: string;
+    onClick?: () => void;
+  }
+}
+
+const symbols = new Map<string, string>();
+
+let symbolId = 0;
+
+// Lazily render the icon in the DOM, so that we can referenced
+// it by id using shadow DOM.
```

Comment:
You can't set bitmap's colors to match current font color as you can with SVG. You can technically change it with CSS using filters, but it's really, really ugly. Sprite maps aren't super beneficial since HTTP/2 minimizes overhead on loading multiple files anyway. One thing we could do is load the SVG via `import`, that would make all the icons pop immediately at the cost of making the initial JS bundle larger, but I'm not sure that's an issue that needs fixing.